File: Spark学习/SparkSQL/src/java/ManuallySpecifyOptions.java
package cn.spark.study.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

public class ManuallySpecifyOptions {
	public static void main(String[] args) {
		SparkConf conf = new SparkConf()
				.setAppName("GenericLoadSave");
		
		JavaSparkContext sc = new JavaSparkContext(conf);
		SQLContext sqlContext = new SQLContext(sc);
		
		DataFrame peopleDF = sqlContext.read().format("json")
				.load("hdfs://spark1:9000/people.json");
		
		peopleDF.select("name").write().format("parquet")
		.save("hdfs://spark1:9000/peopleName_java");
	}
}

File: Spark学习/SparkSQL/src/scala/ManuallySpecifyOptions.scala
package cn.spark.study.sql

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.sql.SQLContext

object ManuallySpecifyOptions {
  def main(args: Array[String]){
    val conf = new SparkConf()
    .setAppName("ManuallySpecifyOptions")
    
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc);
    
    val peopleDF = sqlContext.read.format("json").load("hdfs://spark1:9000/people.json")
    peopleDF.select("name").write.format("paruet").save("hdfs://spark1:9000/people.json")
  }
}